### PR TITLE
JBPM-8840 Open process instance details page from a sub-process

### PR DESCRIPTION
**Pull request by bxf12315**

### CODE REVIEW

**Comment by bxf12315** [comment]
@cristianonicolai @nmirasch It is the first version, please review it.

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/details/ProcessInstanceDetailsTabViewImpl.java [hunk 1]
```diff
@@ -103,8 +104,22 @@
...
+    @DataField("parent-instance-id-link")
+    @SuppressWarnings("PMD.UnusedPrivateField")
+    private HTMLAnchorElement parentInstanceIdLink;
+
+    @Inject
+    @Named("span")
+    @DataField("parent-instance-id")
+    private HTMLElement parentInstanceIdText;
+
+    private String serverTemplateId;
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/details/ProcessInstanceDetailsTabViewImpl.java]: no need to set serverTemplateId

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/details/ProcessInstanceDetailsTabPresenter.java [hunk 2]
```diff
@@ -119,5 +136,9 @@ public void setProcessInstance(ProcessInstanceSummary process) {
         void setParentProcessInstanceIdText(String value);
 
         void setSlaComplianceText(String value);
+
+        void setServerTemplateId(String value);
+
+        void setProcessInstanceDetailsTabPresenter(ProcessInstanceDetailsTabPresenter presenter);
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/details/ProcessInstanceDetailsTabPresenter.java]:
```suggestion
        void setProcessInstanceDetailsCallback(Command callback);
```

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/details/ProcessInstanceDetailsTabPresenter.java [hunk 3]
```diff
@@ -119,5 +136,9 @@ public void setProcessInstance(ProcessInstanceSummary process) {
         void setParentProcessInstanceIdText(String value);
 
         void setSlaComplianceText(String value);
+
+        void setServerTemplateId(String value);
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/details/ProcessInstanceDetailsTabPresenter.java]: no need to have setServerTemplateId

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/details/ProcessInstanceDetailsTabPresenter.java [hunk 4]
```diff
@@ -66,10 +77,12 @@ public void setProcessInstance(ProcessInstanceSummary process) {
         view.setStateText("");
         view.setCurrentActivitiesListBox("");
 
+        view.setProcessInstanceDetailsTabPresenter(this);
         view.setProcessDefinitionIdText(process.getProcessId());
         view.setProcessVersionText(process.getProcessVersion());
         view.setProcessDeploymentText(process.getDeploymentId());
         view.setCorrelationKeyText(process.getCorrelationKey());
+        view.setServerTemplateId(process.getServerTemplateId());
         if (process.getParentId() > 0) {
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/details/ProcessInstanceDetailsTabPresenter.java]:
```suggestion
        if (process.getParentId() > 0) {
        view.setProcessInstanceDetailsCallback(() -> onProcessInstanceIdSelected(process.getServerTemplateId(), process.getDeploymentId(), process.getParentId()));
```
```

**Comment by bxf12315** [comment]
@cristianonicolai updated

**Comment by cristianonicolai** [comment]
Jenkins execute compile downstream build

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Comment by cristianonicolai** [comment]
@bxf12315 when process has no parent, the text should not show as link. Also there is some miss alignment, seems like an extra space before link or text.

Selection_026

Selection_027

**Comment by bxf12315** [comment]
> @bxf12315 when process has no parent, the text should not show as link. Also there is some miss alignment, seems like an extra space before link or text.
> 
> Selection_026
> 
> Selection_027

ok I will try.

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Comment by cristianonicolai** [comment]
@nmirasch @jstastny-cz please take a look.

**Comment by bxf12315** [comment]
> Once the transition from child to parent happens the details correctly show the parent, but in the breadcrumbs there is still ID of the child process, see the video.
> 
> output

I fixed the issue in here https://github.com/kiegroup/jbpm-wb/pull/1408/files#diff-ebe7ff642691153382b30bb8760b45fdR407.

**Comment by cristianonicolai** [comment]
@bxf12315 the fix for that should be in this PR.

**Comment by bxf12315** [comment]
> @bxf12315 the fix for that should be in this PR.

PR 1408 also needs the feature.

**Comment by jstastny-cz** [comment]
In that case I am not approving this one until 1408 is merged.

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

**Comment by jstastny-cz** [comment]
I can still see the issue in the Full Downstream build.
Actually the 1408 didn't fix it (and I did not notice) there. Please fix the breadcrumbs navigation as part of this PR.

**Pull request merged**